aineko/core/dataset.py

```diff
@@ -56,33 +60,43 @@ def __init__(
         node_name: str,
         pipeline_name: str,
         dataset_config: Dict[str, Any],
-        broker: Optional[str] = None,
+        bootstrap_servers: Optional[str] = None,
+        prefix: Optional[str] = None,
+        has_pipeline_prefix: bool = False,
     ):
         """Initialize the consumer."""
-        # Assign dataset name
         self.pipeline_name = pipeline_name
-
-        # Assign kafka config
         self.kafka_config = DEFAULT_KAFKA_CONFIG
+        self.prefix = prefix
+        self.has_pipeline_prefix = has_pipeline_prefix
 
-        # Set consumer parameters
         consumer_config = self.kafka_config.get("CONSUMER_CONFIG")
-
-        # Overwrite broker target if provided
-        if broker:
-            consumer_config["bootstrap.servers"] = broker
+        # Overwrite bootstrap server with broker if provided
+        if bootstrap_servers:
+            consumer_config["bootstrap.servers"] = bootstrap_servers
 
         # Override default config with dataset specific config
         for param, value in dataset_config.get("params", {}).items():
             if param in self.kafka_config.get("CONSUMER_OVERRIDABLES"):
                 consumer_config[param] = dataset_config["params"][value]
 
-        # Add consumer group id based on node name
-        consumer_config["group.id"] = f"{self.pipeline_name}.{node_name}"
+        topic_name = dataset_name
+        if has_pipeline_prefix:
+            topic_name = f"{pipeline_name}.{dataset_name}"
```

Comment:
Why are you using topic_name here instead of dataset_name (or a variant)?

Reply:
Because there is now a separation of the concept of topic name and dataset name. A dataset name is the name within the namespace of the pipeline, so will be the short version that the user uses in pipeline config and node code. The topic name is the actual name used to call the kafka topics, and needs to be in the project namespace, so will contain the pipeline name as a prefix. 
E.g.
a dataset is called `count`
the associated topic is really named `my_pipeline.count`